langchain/src/chat_models/llama_cpp.ts

```diff
@@ -298,11 +317,50 @@ export class ChatLlamaCpp extends SimpleChatModel<LlamaCppCallOptions> {
       .map((message) => {
         let messageText;
         if (message._getType() === "human") {
-          messageText = `[INST] ${message.content} [/INST]`;
+          switch (this.streamingModel) {
+            case "chatML":
+              messageText = `<|im_start|>user\n${message.content}<|im_end|>`;
+              break;
+            case "falcon":
+              messageText = `User: ${message.content}`;
+              break;
+            case "general":
+              messageText = `### Human\n${message.content}`;
+              break;
+            default:
+              messageText = `[INST] ${message.content} [/INST]`;
+              break;
+          }
         } else if (message._getType() === "ai") {
```

Comment:
The next version will support the generation of a response based on chat histories that can have "system", "user" and "model" roles, so I think most of this code could be replaced by the `node-llama-cpp` functionality for this.

I predict that the next beta that includes this functionality will be ready in about ~2 weeks if you're willing to wait for it, but in any case, I think this better be handled by `node-llama-cpp` as it its responsibility to handle the interface against a model.